### PR TITLE
Add key features to config.json

### PR DESCRIPTION
**Pull request by neenjaw**
Key features taken from crystal-lang.org

### CODE REVIEW

**Comment by neenjaw** [comment]
@ErikSchierboom or others

**Review by ErikSchierboom — APPROVED** [review]

File: config.json [hunk 1]
```diff
@@ -604,6 +604,37 @@
...
+      "icon": "TODO"
+    },
+    {
+      "title": "Null Reference Checks",
+      "content": "All types are non-nilable in Crystal, so the compiler checks for null references at compile time.",
+      "icon": "TODO"
+    },
+    {
+      "title": "Concurrency Model",
+      "content": "Crystal has built in concurrency primitives, called fibers, to achieve concurrency. Fibers communicate via channels.",
```
Comment (ErikSchierboom) [inline config.json:625]: This exceeds the maximum of 100 characters.

File: config.json [hunk 2]
```diff
@@ -604,6 +604,37 @@
...
+      "icon": "TODO"
+    },
+    {
+      "title": "C-Interopability",
+      "content": "Crystal has dedicated syntax to easily call native libraries, eliminating the need to reimplement low-level tasks.",
```
Comment (ErikSchierboom) [inline config.json:630]: This exceeds the maximum of 100 characters.

File: config.json [hunk 3]
```diff
@@ -604,6 +604,37 @@
...
+      "icon": "TODO"
+    },
+    {
+      "title": "Macros",
+      "content": "Crystal's answer to metaprogramming is a powerful macro system, which ranges from basic templating to type inspection.",
```
Comment (ErikSchierboom) [inline config.json:635]: This exceeds the maximum of 100 characters.

File: config.json [hunk 4]
```diff
@@ -604,6 +604,37 @@
...
-  "key_features": [],
+  "key_features": [
+    {
+      "title": "Simple Syntax",
+      "content": "Crystal's syntax is heavily inspired by Ruby, so it is natural to read and easy to write.",
+      "icon": "TODO"
+    },
+    {
+      "title": "Static Type Checking",
+      "content": "Crystal is statically type checked with built-in type inference, so type errors are caught early by the compiler.",
```
Comment (ErikSchierboom) [inline config.json:615]: This exceeds the maximum of 100 characters.

**Review by ErikSchierboom — CHANGES_REQUESTED** [review]

File: config.json (hunk 3 above)
Comment (neenjaw) [reply config.json:635]:
```suggestion
      "content": "Crystal's answer to metaprogramming is a powerful macro system.",
```

File: config.json (hunk 2 above)
Comment (neenjaw) [reply config.json:630]:
```suggestion
      "content": "Crystal has dedicated syntax to easily call native libraries, increasing code re-use.",
```

File: config.json (hunk 1 above)
Comment (neenjaw) [reply config.json:625]:
```suggestion
      "content": "Crystal has built in concurrency primitives, called fibers, to achieve concurrency.",
```

File: config.json (hunk 4 above)
Comment (neenjaw) [reply config.json:615]:
```suggestion
      "content": "Crystal is statically type checked with built-in type inference, so type errors are caught early.",
```

**Comment by neenjaw** [comment]
configlet is passing now 🎉

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
Thanks!

**Pull request merged**